tests/test_heartbeat.py

```diff
@@ -162,7 +161,7 @@ def test_heartbeat_can_open_channel_client_close():
     print('SERVER LOST CLIENT :)')
     server_hbchan.close()
     server.close()
-
+    client.close()
```

Comment:
This test `test_heartbeat_can_open_channel_client_close` fails in _Python 3.9_ if the client is closed before the server heartbeat channel is closed. Looking at the other tests, it seemed to me that `client.close()` should be called _after_ `server.close()`.

Reply:
This defeats the purpose of this test. The point is to have the client go away. And the server to detect this by the loss of heartbeats.

Reply:
I see. Then I'll change it back. What I found that the test fails on [python 3.9](https://github.com/shiplu/zerorpc-python/actions/runs/1013103813) if the client is closed before. Any idea why it fails on Python 3.9?

Reply:
The test itself is successful. You can clearly see the message "CLOSE CLIENT SOCKET!!!". Followed by "SERVER LOST CLIENT :)". This is the expected behavior.

Somehow the test is marked as failed. Maybe the assert_raises is not doing its job.

Reply:
Thanks for the confirmation. I'll see if I can fix it!

Reply:
It seems the issue is on nosetests. I run many times these tests on python 3.9. It never fails. Only on github it fails. Then I found one of the tests `` passing with `ok` status.

```
tests.test_buffered_channel.test_congestion_control_server_pushing ... Traceback (most recent call last):
  File "src/gevent/greenlet.py", line 906, in gevent._gevent_cgree nlet.Greenlet.run
  File "/__w/zerorpc-python/zerorpc-python/tests/test_buffered_channel.py", line 429, in server_do
    server_bufchan.emit('coucou', x, timeout=0)  # will fail when x == 1
  File "/usr/local/lib/python3.9/unittest/case.py", line 226, in __exit__
    self._raiseFailure("{} not raised".format(exc_name))
  File "/usr/local/lib/python3.9/unittest/case.py", line 163, in _raiseFailure
    raise self.test_case.failureException(msg)
AssertionError: TimeoutExpired not raised
2021-07-13T21:28:53Z <Greenlet at 0x7efdea2d5260: server_do> failed with AssertionError

ok
```
`AssertionError` and `ok` status together shouldn't be right.

The errors can be seen in my [github action build log](https://github.com/shiplu/zerorpc-python/runs/3061176173?check_suite_focus=true)

So, I think for now we drop supporting python 3.9 (removing the entry from ci.yml and not adding in setup.py). Then probably we can migrate from nosetests to pytest which will enable us to support 3.9 in near future!